pages/room/[...id].js

```diff
@@ -97,21 +103,38 @@ const Room = () => {
         });
 
         socketRef.current.on('return message', (payload) => {
-          console.log(payload.username);
-          console.log(currentUsername);
           setConversation((prevConversation) => {
             return [
               ...prevConversation,
               { text: payload.message, sender: payload.username, fromMe: payload.username == currentUsername },
             ];
           });
         });
+
+        socketRef.current.on('user disconnect', (payload) => {
+          let usersPeerID = [];
+          let participantNames = [];
+          if (payload.users) {
+            usersPeerID = payload.users.map((user) => user.socketID);
+            participantNames = payload.users.map((user) => user.username);
+          }
+          peersRef.current.forEach((peerRef) => {
+            if (usersPeerID.length > 0 && !usersPeerID.includes(peerRef.peerID)) {
+              const removePeerChannelName = peerRef.peer.channelName;
+              setPeers((prevPeers) => prevPeers.filter((peer) => peer.channelName !== removePeerChannelName));
+              setParticipants((prevParticipants) =>
+                prevParticipants.filter((participant) => participantNames.indexOf(participant) >= 0)
+              );
+              peerRef.peer.destroy();
+            }
+          });
+        });
```

Comment:
you can probably use lodash to handle a lot of this logic. Take a look at [difference helper function](https://lodash.com/docs/4.17.15#difference). Make sure to just import the function and not the entire object to optimize performance.